### PR TITLE
Docs: add missing 'memory' field to YAML spec page.

### PR DESCRIPTION
**Pull request by concretevitamin**
Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/reference/yaml-spec.rst [hunk 1]
```diff
@@ -43,26 +43,43 @@ Available fields:
...
-      # Format: <count> (exactly <count> vCPUs) or <count>+
-      # (at least <count> vCPUs).
+      # Format:
+      #   <count>: exactly <count> vCPUs
+      #   <count>+: at least <count> vCPUs
       #
-      # E.g., 4+ would first try to find an instance type with 4 vCPUs. If not
-      # found, it will use the next cheapest instance with more than 4 vCPUs.
-      cpus: 32
+      # E.g., 4+ means first try to find an instance type with 4 vCPUs. If not
```
Comment (cblmemo) [inline docs/source/reference/yaml-spec.rst:70]:
```suggestion
      # E.g., 4+ means first try to find an instance type with >= 4 vCPUs. If not
```

File: docs/source/reference/yaml-spec.rst [hunk 2]
```diff
@@ -179,17 +205,23 @@ Available fields:
     # Values set here can be overridden by a CLI flag:
     # `sky launch/exec --env ENV=val` (if ENV is present).
     #
-    # If you want to use a docker image in a private registry, you can specify your
-    # username, password, and registry server as task environment variable. For example:
+    # If you want to use a docker image in a private registry, you can specify
```
Comment (cblmemo) [inline docs/source/reference/yaml-spec.rst:208]:
```suggestion
    # If you want to use a docker image as runtime environment in a private registry, you can specify
```

**Review by cblmemo — APPROVED** [review]
LGTM! Left 2 nits 🫡

File: docs/source/reference/yaml-spec.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/reference/yaml-spec.rst]: Done

File: docs/source/reference/yaml-spec.rst (hunk 1 above)
Comment (concretevitamin) [reply docs/source/reference/yaml-spec.rst]: Done

**Pull request merged**